.github/workflows/build.yml

```diff
@@ -60,48 +67,103 @@ jobs:
       - name: Install Ninja
         uses: urkle/action-get-ninja@v1
 
+      - name: Download linuxdeploy family
+        if: matrix.app_image == true
+        run: |
+          wget "https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage"
+          wget "https://github.com/linuxdeploy/linuxdeploy-plugin-appimage/releases/download/continuous/linuxdeploy-plugin-appimage-x86_64.AppImage"
+          wget "https://github.com/linuxdeploy/linuxdeploy-plugin-qt/releases/download/continuous/linuxdeploy-plugin-qt-x86_64.AppImage"
+
+      - name: Download JREs for AppImage
+        if: matrix.app_image == true
+        shell: bash
+        run: |
+          ${{ github.workspace }}/.github/scripts/prepare_JREs.sh
+
       - name: Configure CMake
+        if: matrix.app_image != true
         run: |
-          cmake -S . -B build -DCMAKE_INSTALL_PREFIX=install -DCMAKE_BUILD_TYPE=Debug -G Ninja
+          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=${{ env.INSTALL_DIR }} -DCMAKE_BUILD_TYPE=Debug -G Ninja
+
+      - name: Configure CMake for AppImage
+        if: matrix.app_image == true
+        run: |
+          cmake -S . -B ${{ env.BUILD_DIR }} -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Debug -DLauncher_LAYOUT=lin-system -G Ninja
 
       - name: Build
         run: |
-          cmake --build build
+          cmake --build ${{ env.BUILD_DIR }}
 
       - name: Install
+        if: matrix.app_image != true
+        run: |
+          cmake --install ${{ env.BUILD_DIR }}
+
+      - name: Install for AppImage
+        if: matrix.app_image == true
+        run: |
+          DESTDIR=${{ env.INSTALL_DIR }} cmake --install ${{ env.BUILD_DIR }}
+
+      - name: Bundle AppImage
+        if: matrix.app_image == true
+        shell: bash
         run: |
-          cmake --install build
+          export OUTPUT="PolyMC-${{ github.sha }}-x86_64.AppImage"
+          chmod +x linuxdeploy-*.AppImage
+          mkdir -p ${{ env.INSTALL_DIR }}/usr/lib/jvm/java-{8,17}-openjdk
```

Comment:
It feels really weird to put JREs in a JDK directory, are you copying a `jre` folder?

This also reminds me that the LD_LIBRARY_PATH might be wrong because of that. What is the directory structure of the Adoptium JREs?

Reply:
No, the JREs are downloaded directly from Adoptium. Their structures are the same as their JDK alternatives. JRE8 has `bin`, `lib`, `man`, and JRE17 has `bin`, `lib`, `conf` and `legal`.
I have no JRE folders in my `/usr/lib/jvm` even though I have JREs installed, so I put them in the JDK folders like they are on my system.

Reply:
Oh you might be right about the `LD_LIBRARY_PATH`. Let me change that.

Reply:
Only the jre8 path was incorrect in `LD_LIBRARY_PATH`. Fixed it and split up the lines too.